src/main/java/bio/terra/workspace/service/resource/controlled/mappings/ControlledResourceInheritanceMapping.java

```diff
@@ -0,0 +1,91 @@
+ * that entry is not present for private resources.
+ *
+ * <p>This maps workspace roles to the resource roles, which are used for granting cloud
+ * permissions. Sam also has a similar mapping configured for granting Sam permissions. If you
+ * change this map, also check that mapping:
+ * https://github.com/broadinstitute/sam/blob/develop/src/main/resources/reference.conf
+ */
+public class ControlledResourceInheritanceMapping {
+  // Currently, shared resources have the same permission inheritance regardless of whether they
+  // are user- or application-controlled. This list is pulled out separately as a convenience.
+  private static final Multimap<WsmIamRole, ControlledResourceIamRole> SHARED_RESOURCE_MAPPING =
+      new ImmutableMultimap.Builder<WsmIamRole, ControlledResourceIamRole>()
+          .putAll(
+              WsmIamRole.OWNER, ControlledResourceIamRole.EDITOR, ControlledResourceIamRole.WRITER)
+          .putAll(
+              WsmIamRole.WRITER, ControlledResourceIamRole.EDITOR, ControlledResourceIamRole.WRITER)
+          .putAll(WsmIamRole.READER, ControlledResourceIamRole.READER)
+          .build();
+
+  private static final Multimap<WsmIamRole, ControlledResourceIamRole> USER_SHARED_MAPPING =
+      SHARED_RESOURCE_MAPPING;
+
+  private static final Multimap<WsmIamRole, ControlledResourceIamRole> APPLICATION_SHARED_MAPPING =
+      SHARED_RESOURCE_MAPPING;
+
+  // Applications and readers have no permissions on private resources, and MultiMaps ignore keys
+  // without values, so we do not add them here.
+  private static final Multimap<WsmIamRole, ControlledResourceIamRole> USER_PRIVATE_MAPPING =
+      new ImmutableMultimap.Builder<WsmIamRole, ControlledResourceIamRole>()
+          .putAll(
+              WsmIamRole.OWNER,
+              ControlledResourceIamRole.ASSIGNER,
+              ControlledResourceIamRole.EDITOR)
+          .putAll(WsmIamRole.WRITER, ControlledResourceIamRole.EDITOR)
+          .build();
+
+  private static final Multimap<WsmIamRole, ControlledResourceIamRole> APPLICATION_PRIVATE_MAPPING =
+      new ImmutableMultimap.Builder<WsmIamRole, ControlledResourceIamRole>()
+          .putAll(WsmIamRole.OWNER, ControlledResourceIamRole.EDITOR)
+          .putAll(WsmIamRole.WRITER, ControlledResourceIamRole.EDITOR)
+          .build();
+
+  /**
+   * Returns the resource-level roles that each workspace-level role inherits on the specified
+   * category of resource (user-shared, user-private, app-shared, or app-private).
+   *
+   * @return A map whose keys are workspace-level roles and values are the corresponding inherited
+   *     resource-level roles for the specified resource category.
+   */
+  public static Multimap<WsmIamRole, ControlledResourceIamRole> getInheritanceMapping(
+      AccessScopeType accessScope, ManagedByType managedBy) {
+    if (accessScope == AccessScopeType.ACCESS_SCOPE_SHARED) {
```

Comment:
This could be stored as a `Table` as well.

Reply:
It could, though a 4 element table seems a little overkill - I could go either way on this

Reply:
I like smart data structures and dumb code, and I penalize `if` statements.